### ISSUE TITLE
[HOLD for payment 2023-09-20] [$1000] DEV: Share code avatar differs from profile avatar

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#25889]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to the Settings > Share Code.
2. Note the inconsistency in avatars.
## Expected Result: 
Both the profile and Share Code avatars should match
## Actual Result: 
The avatar in the Share Code differs from the one displayed in the profile

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** dev
**Reproducible in staging?:** dev
**Reproducible in production?:** dev
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/1450fb0a-c457-46fb-90a3-898d52aab01a

**Expensify/Expensify Issue URL:**
**Issue reported by:** @niravkakadiya25
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692906813354439

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a91b4da2caec2a9a</li>
        <li>Upwork Job ID: 1695128865636945920</li>
        <li>Last Price Increase: 2023-09-01</li>
<li>Automatic offers: </li>
<ul>
<li>StevenKKC | Contributor | 26516719</li>
<li>niravkakadiya25 | Reporter | 26516722</li>
</ul></li>
    </ul>
</details>

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
DEV: Share code avatar differs from profile avatar 
### What is the root cause of that problem?
Root cause for this problem here is here at this line.
https://github.com/Expensify/App/blob/74be5f627511410373f926c76e50f98c459e93e5/src/libs/UserUtils.js#L151

For `ShareCodePage` we're using `getAvatarUrl` and if it's a default URL we're fetching New Dot avatar. But for the function we're using in `InitialSettingsPage` `getAvatar` is not passing `getDefaultAvatar` 2nd parameter (new dot as boolean) which fetches avatar from old dot.
### What changes do you think we should make in order to solve the problem?

We have to use newdot in both the places.
### What alternative solutions did you explore? (Optional)
NA

**Comment by StevenKKC** [comment]
This is a regression of https://github.com/Expensify/App/pull/25816.
In this PR, does not update [this line](https://github.com/Expensify/App/blob/b38181a2d567cc3cbb5b426f0585db4286747fa6/src/libs/UserUtils.js#L111) in `UserUtils.getDefaultAvatarURL`.

**Comment by StevenKKC** [comment]
## Proposal

### Please state again the problem we are trying to solve in this issue.
Share code avatar differs from profile avatar.

### What is the cause of this issue?
We uses `UserUtils.getAvatar` in profile page, and this function uses `UserUtils.getDefaultAvatar`.
And we uses `UserUtils.getAvatarURL` in share code page, and this function uses `UserUtils.getDefaultAvatarURL`.

In , `UserUtils.getDefaultAvatar` updates the calculation of default avatar, but does not update `UserUtils.getDefaultAvatarURL`.

### What changes do you think we should make to solve the problem?
We should use same calculation in `UserUtils.getDefaultAvatarURL`.

### What alternative solutions have you investigated? (Optional)
None.

**Comment by StevenKKC** [comment]
This only reproducible on DEV, I think not a deploy blocker.

**Comment by luacmartins** [comment]
This is really minor and not a blocker.

**Comment by luacmartins** [comment]
Bump @rushatgabhane

**Comment by situchan** [comment]
This might need to put on hold for #21899

**Comment by b4s36t4** [comment]
I don't think so @situchan. This is different.

**Comment by mallenexpensify** [comment]
@rushatgabhane can you please review the above proposals? Thx

**Comment by mallenexpensify** [comment]
@Christinadobrzyn  I'm off this week, can you please keep 👀 on this then I'll snag it back on Monday?  Thx

**Comment by Christinadobrzyn** [comment]
Hey @rushatgabhane or @mallenexpensify can we remove the `Needs Reproduction` label?

**Comment by luacmartins** [comment]
Agreed! Assigning @StevenKKC

**Comment by StevenKKC** [comment]
@rushatgabhane PR https://github.com/Expensify/App/pull/26858 is ready for review.

**Comment by mallenexpensify** [comment]
Merge freeze is over. The PR should be 🚂 
- https://github.com/Expensify/App/pull/26858

**Comment by luacmartins** [comment]
@mallenexpensify actually, we're trying to stagger merges a bit to avoid an insane checklist in the next deploy. We're prioritizing higher value merges now, and I think this PR can wait a couple of days to be merged.

**Comment by luacmartins** [comment]
That being said, we only had ~25 merged since the last checklist was created. So I went ahead and merged this PR as well.

**Comment by mallenexpensify** [comment]
Re: the urgency bonus, we had a merge freeze and no additional work was needed on the PR from @StevenKKC , so the bonus should be added? Right @luacmartins and @rushatgabhane

**Comment by luacmartins** [comment]
Agreed!

**Comment by StevenKKC** [comment]
@mallenexpensify Could you please make the payment for this issue?

**Comment by luacmartins** [comment]
Bump @StevenKKC

**Comment by luacmartins** [comment]
Just waiting on payment

**Comment by StevenKKC** [comment]
@luacmartins My Upwork account is suspended and now I am solving the issue on my account. Sorry for my delay.

**Comment by mallenexpensify** [comment]
Issue reporter: @niravkakadiya25 paid $50 via Upwork
Contributor: @StevenKKC paid $1500 via Upwork
Contributor+: @rushatgabhane is due $1500

@rushatgabhane please finish the [checklist above](https://github.com/Expensify/App/issues/25889#issuecomment-1734613979)

@StevenKKC , the payment appeared to go through, I'll leave this open for the time being.

**Comment by StevenKKC** [comment]
@mallenexpensify I am afraid to let you know that My Upwork account is suspended so I couldn't withdraw money. I would be really appreciate it if you refund money and send to my friend's account.

**Comment by mallenexpensify** [comment]
@StevenKKC we're currently researching how best to address situations like this. While we do, can you please try to get your Upwork account active again, since that will be the easiest and best way to pay

**Comment by StevenKKC** [comment]
Yes, thank you for your consideration. I received the terrible news from Upwork support that my account has been permanently disabled.

**Comment by JmillsExpensify** [comment]
$1,500 payment approved for @rushatgabhane based on BZ summary.

**Comment by luacmartins** [comment]
Are we all done here?

**Comment by StevenKKC** [comment]
@luacmartins, @mallenexpensify What should I do now? May I receive the payment via my friend's account?

**Comment by StevenKKC** [comment]
@mallenexpensify My friend's Upwork account is fully verified.

**Comment by mallenexpensify** [comment]
I'm actively discussing options with Upwork @StevenKKC , hoping to have an update for you next week, my contact is out this week.

**Comment by luacmartins** [comment]
Still trying to sort payment issues.

**Comment by mallenexpensify** [comment]
Underpaid @niravkakadiya25 so I just bonused $200 and updated the @niravkakadiya25 o reflect the correct price.

**Comment by mallenexpensify** [comment]
@StevenKKC , we're not having much luck, I'm going to leave this open for now, please continue to try to get your Upwork account verified.

**Comment by luacmartins** [comment]
still trying to figure out payment

**Comment by mallenexpensify** [comment]
Bumping to monthly.  @luacmartins , @rushatgabhane , would either of you like to be unassigned?  
The alternative would be to close, but I feel weird about closing without issuing payment, even if we're currently unable to

**Comment by luacmartins** [comment]
Still figuring out payment

**Comment by mallenexpensify** [comment]
Gonna close, I'll get emails if comments are left here and can reopen if/when needed
